app/Http/Controllers/AuthController.php

```diff
@@ -233,25 +233,36 @@ private function registerAccount($request)
      */
     public function deleteUser(Request $request)
     {
-        $user = Auth::user();
+        $user = User::findOrFail($request->id ?? Auth::id());
 
-        $password = $request->input('password');
-        $auth_check = self::verifyCredentials($user->email, $password);
+        if ($user->name != $request->name) {
+            Session::flash('flash_message', 'You need to correctly input the user\'s name before the account is deactivated.');
 
-        if ($user->hasUnpaidOrderlines()) {
-            Session::flash('flash_message', 'You cannot deactivate your account while you have open payments!');
+            return Redirect::back();
+        }
 
-            return Redirect::route('omnomcom::orders::list');
+        if (Auth::id() != $user->id && Auth::user()->cannot('sysadmin')) {
+            Session::flash('flash_message', 'You cannot deactivate someone else\'s account.');
         }
 
-        if ($auth_check == null || $auth_check->id != $user->id) {
-            Session::flash('flash_message', 'You need to provide a valid password to delete your account.');
+        if (Auth::id() == $user->id) {
+            $password = $request->input('password');
+            $auth_check = self::verifyCredentials($user->email, $password);
+            if ($auth_check == null || $auth_check->id != $user->id) {
+                Session::flash('flash_message', 'You need to provide a valid password to delete an account.');
 
-            return Redirect::back();
+                return Redirect::back();
+            }
+        }
+
+        if ($user->hasUnpaidOrderlines()) {
+            Session::flash('flash_message', 'An account cannot be deactivated while it has open payments!');
+
+            return Redirect::route('omnomcom::orders::list');
         }
 
         if ($user->member) {
-            Session::flash('flash_message', 'You cannot deactivate your account while you are a member.');
+            Session::flash('flash_message', 'An account cannot be deactivated while it still has an active membership.');
```

Comment:
I think we should also redirect them to somewhere usefull here (or a message how to deactivate their membership) like we do at the unpaidOrderlines